### DISCUSSION TITLE
POST call on Server Side

### DISCUSSION THREAD

**Discussion by Kurm-U** [vercel/next.js#66555] [App Router]
I want to hit POST API on server side so no one can see in dev tools of browsers.

**Comment by serdar-webdev** [comment]
More details, please

**Reply by Kurm-U** [reply]
Like if I call GET API on server side it's not showing on devtools of browsers and I want to do same thig with POST API call so is that possible.?

**Reply by serdar-webdev** [reply]
code screenshots, why are you using that, explain problem by details

**Reply by Kurm-U** [reply]
Sorry for the delay

Here some Screenshots

![loginjsx](https://github.com/vercel/next.js/assets/170911260/b4525646-9b80-4028-af09-cac700c2d8c7)
in this file I have login handler and this is client component

![image](https://github.com/vercel/next.js/assets/170911260/cd95eac4-e6de-4e22-9165-7297e3bc41d1)
in this file I have created axiosInstance

now whenever this API hit to server it's visible in devtools, and I don't want to show in devtools like other GET API so do you have any solution regarding this...?

Because if we want to do like this we basically create server component and if I turn Login component in server component it doesn't work......

**Reply by serdar-webdev** [reply]
look here how you can solve this:
I don't know how axios works in server, but in next js it recommended to use fetch function, because it's enough ( i think );

here is your server action : 
const login = async (formData: any) => {
  try {
    await fetch("/user/login", {
      method: "POST",
      body: formData,
      headers: {
        "content-type": "multipart/form-data",
        cookie: cookies,
      },
    })
  } catch (error) {
    console.log(error)
  }
}

and here is your component:

const LoginComponent = () => {
  return (
    <div>
      <button
        onclick={async () => {
          login(formData as any)
        }}
      ></button>
    </div>
  )
}

**Reply by serdar-webdev** [reply]
let me know if you have problems with that

**Reply by Kurm-U** [reply]
The problem is if I use fetch then how I change my axiosInstance logic

**Comment by ibharathboga** [comment]
did you look into server actions ? 
server actions are post request type and run in the server. i also think you can use axios there. 
https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations

**Reply by Kurm-U** [reply]
Ya I tried it but I couldn't grasp the logic so I'm looking for more example.

**Reply by ibharathboga** [reply]
there are lots of youtube videos explaining it. this one helped me : [youtube](https://www.youtube.com/watch?v=W6V91VeghjI)
maybe watch this video and try again ? if you still face issues, feel free to ping me

**Reply by Kurm-U** [reply]
thanks for suggestion let me check...!

**Comment by serdar-webdev** [comment]
Here is the server action ( it will run in server and users dont have access to it ), here you have to try to implement axios, i think it will work

![code1](https://github.com/vercel/next.js/assets/115923465/8737b2e5-b786-4df2-b732-0cb90278ed33)

Here your client component

![code](https://github.com/vercel/next.js/assets/115923465/b1da5a6b-cb72-4afb-bdf1-3e6136ecb122)

**Reply by Kurm-U** [reply]
I know it's late but can you tell me about file structure.. cause I think my problem is file structure

**Reply by serdar-webdev** [reply]
There is nothing specific;
I use app router;
src ( root )
  - app --> app's pages here ( app router )
  - actions  --> my server actions

**Reply by Kurm-U** [reply]
is this correct structure

![image](https://github.com/vercel/next.js/assets/170911260/f4c7c033-16f8-4568-951a-c3035191a810)

**Reply by serdar-webdev** [reply]
it won't be a problem, tell me what version of next do u use and which directory, app or pages?
So i dont use pages folder; here what i usually have inside src folder:
- app
- components
- actions
- store
and etc

**Reply by Kurm-U** [reply]
I'm using Next 14.2.3 and I didn't work with Next before so I don't have deep idea about app or pages routing.

**Reply by serdar-webdev** [reply]
So you are using app router, it recommended and learn about server actions in next js and how to handle them, so after you can use my code snippets above and try it yourself; https://www.youtube.com/watch?v=5MRLO-7O2So

**Reply by Kurm-U** [reply]
Okay let me check
